src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -76,21 +77,21 @@ public IterableOf(final List<X> list) {
      * @param list The list
      * @since 0.21
      */
-    public IterableOf(final Iterator<X> list) {
+    public IterableOf(final Iterator<? extends X> list) {
         this(() -> list);
     }
 
     /**
      * Ctor.
      * @param sclr The encapsulated iterator of x
      */
-    public IterableOf(final Scalar<Iterator<X>> sclr) {
+    public IterableOf(final Scalar<Iterator<? extends X>> sclr) {
         this.itr = sclr;
     }
 
     @Override
     public Iterator<X> iterator() {
-        return new Unchecked<>(this.itr).value();
+        return new Mapped<>(x -> x, new Unchecked<>(this.itr).value());
```

Comment:
@andreoss Has been needed put a `Mapped` here? Why? An `IteratorOf` can be used instead?

Reply:
@fabriciofx `Mapped` is used in order to coerce `? extends X` to `X`. `IteratorOf` wouldn't be enough for that

Reply:
@andreoss please add a todo so that this can be investigated and the code simplified, there must be some way to do something simpler.